### PR TITLE
add journalling endpoints

### PR DESCRIPTION
**Pull request by UwaoEvan**
**What i did**
- Migrated the journaling endpoints.

### CODE REVIEW

File: src/routes/journal/index.ts [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+  fastify.get(
+    "/categories",
+    {
+      schema: {
+        response: {
+          200: JournalCategories,
+        },
+      },
+    },
+    async () => journalCategories,
```
Comment (tindNan) [inline src/routes/journal/index.ts:73]: prefer this:
```
(_, reply) =>  reply.send(journalCategories)
```

File: src/routes/journal/index.ts [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+          `Journal with the id ${journal_id} not found.`,
+        );
+      }
+
+      return updatedJournalEntry;
+    },
+  );
+
+  fastify.delete<{ Params: JournalEntryParams }>(
+    "/:journal_id",
```
Comment (tindNan) [inline src/routes/journal/index.ts:266]: missing param definition

File: src/routes/journal/index.ts [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+      }
+
+      return {};
+    },
+  );
+};
+
+export default journalRouter;
+
+const shuffleJournals = (array: string[], seed: string) => {
```
Comment (tindNan) [inline src/routes/journal/index.ts:293]: would be nice to include a test for this function

File: src/routes/journal/index.ts [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+
+      return updatedJournalEntry;
+    },
+  );
+
+  fastify.delete<{ Params: JournalEntryParams }>(
+    "/:journal_id",
+    async (request) => {
+      const { journal_id } = request.params;
+      const journalEntry = await fastify.db
```
Comment (tindNan) [inline src/routes/journal/index.ts:269]: please destructure this

File: src/routes/journaling/index.ts [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+  [category: string]: {
+    [subCategory: string]: string[];
+  };
+}
+
+const JournalEntry = Type.Object({
+  id: Type.String(),
+  createdAt: Type.Optional(Type.String({ format: "date-time" })),
+  updatedAt: Type.Optional(Type.String({ format: "date-time" })),
+  deletedAt: Type.Optional(Type.String({ format: "date-time" })),
```
Comment (tindNan) [inline src/routes/journaling/index.ts:19]: does the python version use camelCase or snake_case for this?

File: src/routes/journaling/index.ts [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+
+const shuffleJournals = (array: string[], seed: string) => {
+  const rng = seedrandom(seed);
+  const shuffledArray = array.slice();
+  for (let i = shuffledArray.length - 1; i > 0; i--) {
+    let j = Math.floor(rng() * (i + 1));
+    [shuffledArray[i], shuffledArray[j]] = [shuffledArray[j], shuffledArray[i]];
+  }
+  return shuffledArray.slice(0, 3);
+};
```
Comment (tindNan) [inline src/routes/journaling/index.ts:306]: very minor, but can we move this to the top of the file?

**Comment by UwaoEvan** [comment]
I was thinking of extracting it to a separate file since affirmations also
will need to use this shuffling method.

File: src/routes/journaling/index.ts (hunk 6 above)
Comment (tindNan) [reply src/routes/journaling/index.ts:306]: and I wonder if this is a function that's used elsewhere e.g. daily affirmations etc 

if yes then we might need to give it a more generic name like `shuffleArray`

**Comment by UwaoEvan** [comment]
Yes. We can discuss about this tomorrow.

**Comment by tindNan** [comment]
@UwaoEvan pending tests

We could create a simple test for one of the POST endpoints and we can use that as a template

**Pull request merged**